### PR TITLE
[ci] Add expectations check for PR branch

### PR DESCRIPTION
**Pull request by sbc100**
See #23146

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,5 +64,19 @@ jobs:
             echo "Test expectations are out-of-date on the target branch."
             echo "You can run `./tools/maint/rebaseline_tests.py --new-branch`"
             echo "and use it to create a seperate PR."
+            echo "-- This failure is only a warning and can be ignored"
             exit 1
           fi
+      - name: Check test expectations on PR branch
+        run: |
+          git checkout ${{ github.base_name }}
```
Comment (dschuff) [inline .github/workflows/ci.yml:72]: what is gitub.base_name? I don't see it on https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs

Comment (sbc100) [reply .github/workflows/ci.yml:72]: Oops, should be base_ref

**Comment by kripken** [comment]
Still failing on CI?

**Comment by sbc100** [comment]
Looks like it working now

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -64,5 +65,23 @@ jobs:
...
           fi
+      - name: Check test expectations on PR branch
+        run: |
+          echo "Checking out ${{ github.ref }} (${{ github.sha }})"
+          # For some reason we cannot pass ${{ github.ref }} direclty to git
+          # since it doesn't recognise it.
+          git checkout ${{ github.sha }}
+          git rev-parse HEAD
+          ./bootstrap
+          if ! ./tools/maint/rebaseline_tests.py --check-only; then
```
[the remark this answers is not in the document]
Comment (sbc100) [reply .github/workflows/ci.yml:79]: Hmm.. you are correct, there could be an emscripten-side change that causes one of the native libraries to change.

[the remark this answers is not in the document]
Comment (sbc100) [reply .github/workflows/ci.yml]: Done

**Review by dschuff — APPROVED** [review]

**Pull request merged**